### DISCUSSION TITLE
useController + checkbox input + text value

### DISCUSSION THREAD

**Discussion by csantos-nydig** [react-hook-form/react-hook-form#7225] [Q&A]
### Version Number

7.21.0

### Codesandbox/Expo snack

https://codesandbox.io/s/form-checkbox-dy2kl?file=/src/App.tsx

### Steps to reproduce

1. Check the first checkbox (uses `register`)
2. Click on first 'Submit'
3. See the value "YES" for the `isDeveloper` field in the browser alert
4. Check the second checkbox (uses `useController`)
5. Click on second 'Submit'
6. See the value 'true' for the `isDeveloper` field in the browser alert. I am expecting to get **"YES"**

### Expected behaviour

I'd expect to get "YES" for both checkboxes no matter if I use `register` or `useController`

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Edge

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
regsiter: uncontrolled which we are retrieve the `value` on your behavior
useController: controlled which by default is getting `e.target.checked`.

**Reply by csantos-nydig** [reply]
Sorry @bluebill1049 - could you elaborate in your answer please

**Reply by bluebill1049** [reply]
- register (uncontrolled) returns its value if you have provided, otherwise boolean: true/false

```tsx
<input name="checkbox" value="test" />
<input name="checkbox"  />
```

- controlled component is by what you provided

```tsx
<input type="checkbox" onChange={field.onChange} /> /> // default checked will return
<input type="checkbox" onChange={e => field.onChange(e.target.value)} /> /> // what you provided to onChange will return
```

why the controlled component does not look into the `value` then, because the controlled component is really flexible and the user's input value is hosted into its local state
